src/app/qgsrecentprojectsmenueventfilter.cpp

```diff
@@ -0,0 +1,77 @@
+/***************************************************************************
+               qgsrecentprojectsmenueventfilter.cpp
+               ----------------------------------------------------
+    begin                : August 2023
+    copyright            : (C) 2023 by Yoann Quenach de Quivillic
+    email                : yoann dot quenach at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsrecentprojectsmenueventfilter.h"
+
+#include "qgswelcomepage.h"
+#include "qgsprojectlistitemdelegate.h"
+
+#include <QMenu>
+#include <QAction>
+#include <QEvent>
+#include <QMouseEvent>
+
+
+QgsRecentProjectsMenuEventFilter::QgsRecentProjectsMenuEventFilter( QgsWelcomePage *welcomePage, QObject *parent )
+  : QObject( parent ), mWelcomePage( welcomePage )
+{
+}
+
+bool QgsRecentProjectsMenuEventFilter::eventFilter( QObject *obj, QEvent *event )
+{
+
+  if ( event->type() != QEvent::MouseButtonPress )
+    return false;
+
+  QMouseEvent *mouseEvent = dynamic_cast<QMouseEvent *>( event );
+  if ( !mouseEvent )
+    return false;
+
+  if ( mouseEvent->button() != Qt::RightButton )
+    return false;
+
+  QMenu *menu = qobject_cast<QMenu *>( obj );
+  if ( !menu )
+    return false;
+
+  QAction *action = menu->actionAt( mouseEvent->pos() );
+  if ( !action )
+    return false;
+
+  int actionIndex = menu->actions().indexOf( action );
+
+  // The last action is the "Clear recent projects" action
+  if ( actionIndex == menu->actions().count() - 1 )
```

Comment:
This is a little fragile, as the menu may be reworked in future and additional non-project items added. A safer approach would be to set user data on the project actions and check that here.

Reply:
Agreed